Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,552 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <API/ToolsApplicationAPI.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzCore/Utils/Utils.h>
+#include <Entity/PrefabEditorEntityOwnershipInterface.h>
+#include <Prefab/PrefabSystemComponentInterface.h>
+#include <Viewport/ViewBookmarkLoader.h>
+
+namespace AzToolsFramework
+{
+    static constexpr const char* s_viewBookmarksRegistryPath = "/O3DE/ViewBookmarks/";
+    static constexpr const char* s_localBookmarksKey = "LocalBookmarks";
+    static constexpr const char* s_lastKnownLocationKey = "LastKnownLocation";
+
+    //Temporary value until there UI to expose the fields.
+    static constexpr int s_numOfDefaultLocationsInLevel = 12;
```

Comment:
update: Please update these to use `UpperCamelCase` convention (the `s_` prefix is reserved for static class members)